File: Classes/Expressions/Operations/AJRModulusOperator.m

#import "AJRModulusOperator.h"

@implementation AJRModulusOperator

- (id)performOperatorWithLeft:(id)left andRight:(id)right error:(NSError **)error
{
    return [NSNumber numberWithLong:[left longValue] % [right longValue]];
}

@end

File: Classes/Expressions/Operations/AJRModulusOperator.swift
//
//  ModulusOperator.swift
//  radar-core
//
//  Created by Alex Raftis on 8/13/18.
//

import Foundation

@objcMembers
open class AJRModulusOperator : AJROperator, AJRIntegerOperator {
    
    public func performIntOperator(withLeft left: Int, andRight right: Int) throws -> Any? {
        if right == 0 {
            throw AJRFunctionError.invalidArgument("Attempt to divide by 0.")
        }
        return left % right
    }
    
}
